### PR TITLE
fix(side drawer): dark mode token updates

### PR DESCRIPTION
**Pull request by wbarbee**
## Summary

Side Drawer dark mode implementation implements new color token. The only real change in Figma file is containing the `.body` text of the <dialog> in a new, distinct `Background/Accent/Subtle` background color.

## ADO Story or GitHub Issue Link

[Side Drawer User Story](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/2003575)

## Figma Link

[Side Drawer Dev-Ready component](https://www.figma.com/design/CQuDZEeLiuGiALvCWjAKlu/branch/qMpff4GuFUEcsMUkvacS3U/Applications---Component-Library?node-id=8451-17321&node-type=canvas&m=dev)

## Notes

*** dependency on Button component styling from foundation [[PR](https://github.com/kyndryl-design-system/shidoka-foundation/pull/63)] 

## To Do
need to add  `--kd-background-blur-medium` as a manual token in foundation

## Checklist

- [x] Used Conventional Commit messages as outlined in the contributing guide.
  - [x] Noted breaking changes (if any).
- [x] Documented/updated all props, events, slots, parts, etc with JSDoc.
  - [x] Ran the `analyze` command to update Storybook docs.
- [x] Added/updated Stories with controls to cover all variants.
- [x] Ran `test` locally to address any failures.
- [x] Added/updated the Figma link for the Story's Design tab.
- [x] Added any new component exports to the src/index.ts file

## Screenshots

*** NOTE: button stylings will be corrected when `<kd-button/>` tokens are updated and merged into `next` branch
Screenshot 2024-12-10 at 9 28 00 AM
Screenshot 2024-12-10 at 9 27 39 AM

### CODE REVIEW

File: src/components/reusable/sideDrawer/sideDrawer.scss [hunk 1]
```diff
@@ -86,9 +73,11 @@ form {
 
 .body {
   margin: 0 32px;
-  padding-right: 24px;
+  padding: 32px;
   flex-grow: 1;
   overflow-y: auto;
+  color: var(--kd-color-text-level-secondary);
+  background-color: var(--kd-color-background-accent-subtle);
```
Comment (brian-patrick-3) [inline src/components/reusable/sideDrawer/sideDrawer.scss:80]: I don't think we actually want a background color on the body here. I think it's there in Figma just to indicate it's a slot. Can double-check with Chadan, but it seems off to me.

File: src/components/reusable/sideDrawer/sideDrawer.scss [hunk 2]
```diff
@@ -164,7 +154,7 @@ h1 {
     transform: translateX(100%);
 
     &::backdrop {
-      background-color: rgba(0, 0, 0, 0);
+      background: var(--kd-color--neutral-dark-opacity-50, rgba(61, 60, 60, 0.5));
```
Comment (brian-patrick-3) [inline src/components/reusable/sideDrawer/sideDrawer.scss:157]: Don't think we should really need a fallback value for the variable here.

File: src/components/reusable/sideDrawer/sideDrawer.scss [hunk 3]
```diff
@@ -174,15 +164,18 @@ h1 {
       transform: none;
 
       &::backdrop {
-        background-color: rgba(0, 0, 0, 0.4);
+        background: var(--kd-color--neutral-dark-opacity-50, rgba(61, 60, 60, 0.5));
+        backdrop-filter: blur(0.10000000149011612px);
```
Comment (brian-patrick-3) [inline src/components/reusable/sideDrawer/sideDrawer.scss:168]: Curious, where did this blur come from?

File: src/components/reusable/sideDrawer/sideDrawer.scss [hunk 4]
```diff
@@ -9,9 +9,9 @@ dialog {
   margin: 0;
   padding: 0;
   color: inherit;
-  background: var(--kd-color-background-ui-default);
-  border: 1px solid var(--kd-color-border-light);
-  box-shadow: 0px 0px 40px 0px rgba(0, 0, 0, 0.12);
+  background: var(--kd-color-background-container-default);
+  border: none;
+  box-shadow: 0px 2px 12px 0px rgba(61, 60, 60, 0.25);
```
Comment (brian-patrick-3) [inline src/components/reusable/sideDrawer/sideDrawer.scss:14]: We should probably use the correct Elevation mixin here instead.

**Review by brian-patrick-3 — CHANGES_REQUESTED** [review]

File: src/components/reusable/sideDrawer/sideDrawer.scss (hunk 2 above)
Comment (brian-patrick-3) [reply src/components/reusable/sideDrawer/sideDrawer.scss:157]: There's also an extra dash after color

File: src/components/reusable/sideDrawer/sideDrawer.scss (hunk 3 above)
Comment (wbarbee) [reply src/components/reusable/sideDrawer/sideDrawer.scss]: rounded up to 1px, blur is more prominent now
Screenshot 2024-12-04 at 9 26 55 AM

Comment (brian-patrick-3) [reply src/components/reusable/sideDrawer/sideDrawer.scss]: I [left a comment in Figma](https://www.figma.com/design/qMpff4GuFUEcsMUkvacS3U?node-id=19910-110621&m=dev#1051618379), let's see what they say.

**Review by srpriyankashetty — APPROVED** [review]
Verified Side drawer and side drawer before close in dark mode theme and looks fine.
Verified the background blur, when side drawer is open in both the theme and looks fine.

As mentioned by Dev, Styling and disable submit button gray out - hope will be taken care in kd-button fixes.

File: src/components/reusable/sideDrawer/sideDrawer.scss [hunk 5]
```diff
@@ -86,9 +74,10 @@ form {
 
 .body {
   margin: 0 32px;
-  padding-right: 24px;
+  padding: 32px;
```
Comment (brian-patrick-3) [inline src/components/reusable/sideDrawer/sideDrawer.scss:77]: The body text is no longer aligned with the title.

File: src/components/reusable/sideDrawer/sideDrawer.scss [hunk 6]
```diff
@@ -86,9 +74,10 @@ form {
...
   flex-grow: 1;
   overflow-y: auto;
+  color: var(--kd-color-text-level-secondary);
```
Comment (brian-patrick-3) [inline src/components/reusable/sideDrawer/sideDrawer.scss:80]: Might want to double-check on this, as it fails contrast accessibility.

File: src/components/reusable/sideDrawer/sideDrawer.scss [hunk 7]
```diff
@@ -174,15 +164,18 @@ h1 {
       transform: none;
 
       &::backdrop {
-        background-color: rgba(0, 0, 0, 0.4);
+        background: var(--kd-color-background-opacity-4);
+        backdrop-filter: blur(calc(var(--kd-blur-medium, 4px) / 2));
```
Comment (brian-patrick-3) [inline src/components/reusable/sideDrawer/sideDrawer.scss:168]: This should really just be a token without any calc.

**Review by brian-patrick-3 — CHANGES_REQUESTED** [review]

File: src/components/reusable/sideDrawer/sideDrawer.scss (hunk 5 above)
Comment (wbarbee) [reply src/components/reusable/sideDrawer/sideDrawer.scss:77]: ✅

File: src/components/reusable/sideDrawer/sideDrawer.scss (hunk 7 above)
Comment (wbarbee) [reply src/components/reusable/sideDrawer/sideDrawer.scss]: naming this `--kd-background-blur-medium` to be added as a manual token

File: src/components/reusable/sideDrawer/sideDrawer.scss (hunk 6 above)
Comment (wbarbee) [reply src/components/reusable/sideDrawer/sideDrawer.scss:80]: i'm not seeing the accessibility errors:
Screenshot 2024-12-10 at 9 46 41 AM

Screenshot 2024-12-10 at 9 47 14 AM

Comment (brian-patrick-3) [reply src/components/reusable/sideDrawer/sideDrawer.scss:80]: Ah, it was in light mode

Comment (wbarbee) [reply src/components/reusable/sideDrawer/sideDrawer.scss:92]: i see, you need to toggle light or dark with the drawer open for the violations to appear. got them down to just button violations now that should be sorted with the updated button component

Screenshot 2024-12-10 at 10 09 28 AM

**Review by brian-patrick-3 — APPROVED** [review]

**Comment by tulasigudibanda** [comment]
@wbarbee  , Chandan added some styles for the X button states in modal. I think they may be applicable for side drawer close button too.

**Comment by tulasigudibanda** [comment]
This color token is not present in foundation : box-shadow: var(--kd-elevation-level-3);

**Pull request merged**